packages/dds/tree/src/shared-tree/schematizingTreeView.ts

```diff
@@ -301,73 +300,61 @@ export class SchematizingSimpleTreeView<
 	 * If implicit initialization (or some other edit) is desired, it should be done outside of this method.
 	 */
 	private update(): void {
-		this.disposeView();
+		this.disposeFlexView();
 
 		const compatibility = this.viewSchema.checkCompatibility(this.checkout.storedSchema);
 
-		let lastRoot =
-			this.compatibility.canView && this.view !== undefined ? this.root : undefined;
+		let lastRoot: ReadableField<TRootSchema> | undefined;
 		this.currentCompatibility = {
 			...compatibility,
 			canInitialize: canInitialize(this.checkout),
 		};
 
+		const anchors = this.checkout.forest.anchors;
+		const slots = anchors.slots;
+
 		if (compatibility.canView) {
 			// Trigger "rootChanged" if the root changes in the future.
 			// Currently there is no good way to do this as FlexTreeField has no events for changes.
 			// this.view.flexTree.on(????)
 			// As a workaround for the above, trigger "rootChanged" in "afterBatch"
 			// which isn't the correct time since we normally do events during the batch when the forest is modified, but its better than nothing.
 			// TODO: provide a better event: this.view.flexTree.on(????)
-			const cleanupCheckOutEvents = this.checkout.events.on("afterBatch", () => {
-				// In the initialization flow, this event is raised before the correct compatibility w.r.t the new schema is calculated.
-				// Accessing `this.root` in that case can throw. It's OK to ignore this because:
-				// - The rootChanged event will already be raised at the end of the current upgrade
-				// - It doesn't matter that `lastRoot` isn't updated in this case, because `update` will be called again before the upgrade
-				//   completes (at which point this callback and the `lastRoot` captured here will be out of scope anyway)
-				if (!this.midUpgrade && lastRoot !== this.root) {
-					lastRoot = this.root;
-					this.events.emit("rootChanged");
-				}
-			});
-
-			const onViewDispose = (): void => {
```

Comment:
this logic is no longer triggered by dispose of the context, bit instead run in disposeFlexView after disposing the Context. The same is true for the on dispose logic added by requireSchema below